### PR TITLE
[14925] Add interfaces for ReadCondition

### PR DESCRIPTION
**Pull request by MiguelCompany**
## Description

This PR adds the necessary changes to the public API in order to support ReadCondition.

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- **N/A** Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added. 
- [x] Any new/modified methods have been properly documented using Doxygen. 
- [ ] Fast DDS test suite has been run locally. 
- [ ] Changes are ABI compatible. 
- [ ] Changes are API compatible. <-- Note: we are breaking the API on unsupported methods
- [ ] Documentation builds and tests pass locally. 
- **N/A** New feature has been added to the `versions.md` file (if applicable).
- **N/A** New feature has been documented/Current behavior is correctly described in the documentation. 

## Reviewer Checklist
- [ ] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Comment by MiguelCompany** [comment]
@richiprosima Please test this

**Comment by MiguelBarro** [comment]
According with **DDS 1.4 2.2.2.5.3.10** the following `DataReader` methods should fail if the given condition is not attach to the reader. Thus, they cannot neither default nor accept a `nullptr`:
+ read_w_condition()
+ take_w_condition()
+ read_next_instance_w_condition()
+ take_next_instance_w_condition()

**Comment by MiguelCompany** [comment]
@richiprosima Please test this

File: include/fastdds/dds/subscriber/DataReader.hpp [hunk 1]
```diff
@@ -478,16 +478,16 @@ class DataReader : public DomainEntity
...
-     * @param[in]     a_condition     A ReadCondition that returned @c sample_states must pass
+     * @param[in]     a_condition     A ReadCondition that returned @c data_values must pass
      *
      * @return Any of the standard return codes.
      */
     RTPS_DllAPI ReturnCode_t read_next_instance_w_condition(
             LoanableCollection& data_values,
             SampleInfoSeq& sample_infos,
-            int32_t max_samples = LENGTH_UNLIMITED,
-            const InstanceHandle_t& previous_handle = HANDLE_NIL,
```
Comment (MiguelBarro) [inline include/fastdds/dds/subscriber/DataReader.hpp:489]: These defaults are acceptable and desirable

File: include/fastdds/dds/subscriber/DataReader.hpp [hunk 2]
```diff
@@ -570,15 +570,15 @@ class DataReader : public DomainEntity
...
      *                                @ref LENGTH_UNLIMITED is provided, as many samples will be returned as are.
-     * @param[in]     a_condition     A ReadCondition that returned @c sample_states must pass
+     * @param[in]     a_condition     A ReadCondition that returned @c data_values must pass
      *
      * @return Any of the standard return codes.
      */
     RTPS_DllAPI ReturnCode_t take_w_condition(
             LoanableCollection& data_values,
             SampleInfoSeq& sample_infos,
-            int32_t max_samples = LENGTH_UNLIMITED,
```
Comment (MiguelBarro) [inline include/fastdds/dds/subscriber/DataReader.hpp:580]: This default is acceptable and desirable

File: include/fastdds/dds/subscriber/DataReader.hpp [hunk 3]
```diff
@@ -682,16 +682,16 @@ class DataReader : public DomainEntity
...
-     * @param[in]     a_condition     A ReadCondition that returned @c sample_states must pass
+     * @param[in]     a_condition     A ReadCondition that returned @c data_values must pass
      *
      * @return Any of the standard return codes.
      */
     RTPS_DllAPI ReturnCode_t take_next_instance_w_condition(
             LoanableCollection& data_values,
             SampleInfoSeq& sample_infos,
-            int32_t max_samples = LENGTH_UNLIMITED,
-            const InstanceHandle_t& previous_handle = HANDLE_NIL,
```
Comment (MiguelBarro) [inline include/fastdds/dds/subscriber/DataReader.hpp:693]: These defaults are acceptable and desirable

**Review by MiguelBarro — CHANGES_REQUESTED** [review]
Keep the reasonable defaults

**Review by MiguelBarro — APPROVED** [review]
LGTM
It's a pity to be forced to remove all default argument values to keep the standard parameter order 🙄

**Comment by MiguelCompany** [comment]
@richiprosima Please test mac

**Pull request merged**